docs/designs/link-output.md

```diff
@@ -0,0 +1,97 @@
+# Link And Bookmarks Output
+
+Docfx generate `links` and `bookmarks` information of each files into one single file `links.json` for future data utilization of downstream service of docfx.
+
+## Links
+
+`links` includes all the links which has been processed by docfx:
+
+- Relative link in source content will be processed by docfx and convert it to absolute link if it's valid, otherwise keep it untouched.
+
+- Absolute link in source content will NOT processed by docfx
+
+| Links in source content | Examples              | Output in `links.json` |
+| ----------------------- | --------------------- | ---------------------- |
+| Relative Link           | ./a.md (valid)        | /basepath/article/a    |
+|                         | a/b.md (invalid)      | a/b.md                 |
+| Absolute Link           | /basepath/articles/b  | /basepath/articles/b   |
+|                         | www.github.com/x      | www.github.com/x       |
+|                         | https://docs.com/y    | https://docs/com/y     |
+
+> NOTE: the absolute links converted from relative links by docfx don't contains `locale` and `versioning` information
+
+##  Bookmarks
+
+`bookmarks` includes all bookmarks ids generated/defined from source content, includes its inclusions(token/code-snippets)
```

Comment:
What about bookmarks ids generated/defined from template? Do we need to include or exclude them? I think Ying is confirming with Adam about whether template can export this.

Reply:
We don't need generate bookmark info anymore. Deleted this part